### ISSUE TITLE
Improve create message route

### ISSUE THREAD

**Issue by A5rocks** [discord/discord-api-spec#1]
Currently, the create message route is... kinda sucky:

https://github.com/discord/discord-api-spec/blob/bab7bf1988b7a49fb8f45fae936126122d0a25e4/specs/openapi.json#L4496-L4537

This could be better expressed as:

```json
"patternProperties": {
  "^files\[[0-9]+\]$": {
    "type": "string",
    "contentEncoding": "binary"
  }
}
```

using [`patternProperties`](https://json-schema.org/draft/2020-12/json-schema-core.html#name-patternproperties).

**Comment by IanMitchell** [comment]
🤔 pretty sure the ID can be any snowflake as well

**Comment by A5rocks** [comment]
Yeah this uses a regex to say the number is any sequence of digits. I would use `\d` but that's not in the restricted subset of regex JSON schema recommends :(

**Comment by infinitestory** [comment]
This is a technical limitation needed to express that we don't accept unlimited uploads in a single request.  (In general, for requests, I'd prefer that the spec expresses a subset rather than a superset of allowed inputs, if it can't be exactly precise).

**Comment by A5rocks** [comment]
How about using `maxProperties` then? IIRC you can only send the `payload_json` key for the form body (... maybe I'm misremembering?) so just set the max number of properties to the max amount of files + 1? (... or maybe without the +1?).

**Comment by infinitestory** [comment]
i think we currently accept body properties as form properties (but if payload_json is present, they all have to be in there instead)?  i could be wrong as well

**Comment by A5rocks** [comment]
I checked, you're correct. Maybe there should be an `oneOf` where one branch allows only these fixed names and no `payload_json`, and the other branch allows any fixed names and only an extra `payload_json` field?
